gips/data/landsat/landsat.py

```diff
@@ -291,6 +294,60 @@ def __init__(self, filename):
             raise Exception("Sensor %s not supported: %s" % (self.sensor, filename))
         self._version = self.version
 
+    def filter(self, pclouds=100, **kwargs):
+        """
+        Filters current asset, currently based on cloud cover.
+
+        pclouds is a number between 0 and 100.
+
+        kwargs is reserved for future filtering parameters.
+        """
+        if pclouds >= 100:
+            return True
+
+        if os.path.exists(self.filename):
+            mtlfilename = self.extract([f for f in self.datafiles() if f.endswith('MTL.txt')])[0]
+            with utils.error_handler('Error reading metadata file ' + mtlfilename):
+                with open(mtlfilename, 'r') as mtlfile:
+                    text = mtlfile.read()
+                cc_pattern = r".*CLOUD_COVER = (\d+.?\d*)"
+                cloud_cover = re.match(
+                    cc_pattern,
+                    text,
+                    flags=re.DOTALL
+                )
+                if not cloud_cover:
+                    raise ValueError("No match for '{}' found in {}".format(cc_pattern, mtlfilename))
+                scene_cloud_cover = float(cloud_cover.group(1))
+        else:
+            api_key = api.login(
+                self.Repository.get_setting('username'),
+                self.Repository.get_setting('password')
+            )
+            dataset_name = landsatAsset._sensors[self.sensor]['ee_dataset']
+            path_field = landsatAsset._ee_datasets[dataset_name]['path_field']
+            row_field = landsatAsset._ee_datasets[dataset_name]['row_field']
+            response = api.search(
+                dataset_name, 'EE',
+                where={
+                    path_field: self.tile[0:3], row_field: self.tile[3:]},
+                start_date=datetime.strftime(self.date, "%Y-%m-%d"),
+                end_date=datetime.strftime(self.date, "%Y-%m-%d"),
+                api_key=api_key
+            )
+            metadata = requests.get(
+                response['data']['results'][0]['metadataUrl']
+            ).text
+            xml = ElementTree.fromstring(metadata)
+            # Indexing an Element instance returns it's children
+            scene_cloud_cover_el = xml.find(
+                ".//{http://earthexplorer.usgs.gov/eemetadata.xsd}metadataField[@name='Scene Cloud Cover']"
+            )[0]
+
+            scene_cloud_cover = float(scene_cloud_cover_el.text)
```

Comment:
To keep things moving, I'm going to say this fits the bill for now, but the retrieval/digestion of the metadata should probably be moved to an instance method `landsatAsset.get_metadata`, that uses this bit of xml parsing to populate values (either instance attributes or fields in a metadata dictionary).  Moving it to being a data member will make it more generally useful.  For example, I test-drove this PR by the following:
```
In [13]: inv = lsd.inventory(pclouds=1)

In [14]: inv.numfiles
Out[14]: 0

In [15]: inv = lsd.inventory(pclouds=10)

In [16]: inv.numfiles
Out[16]: 2

In [17]: inv = lsd.inventory(pclouds=5)

In [18]: inv.numfiles
Out[18]: 1
```
But I had to find those values by bisection because the `scene_cloud_cover` isn't exposed.